cmd/draft/init.go

```diff
@@ -194,6 +219,91 @@ func (i *initCmd) setupDraftd() error {
 	return nil
 }
 
+// setupDriverEnv prepares os.Env for the driver. It operates on os.Env because
+// the plugin subsystem itself needs access to the environment variables
+// created here. This extends setupPluginEnv by giving init drivers additional
+// context on what feature flags were set.
+func (i *initCmd) setupDriverEnv() {
+	envMap := map[string]string{
+		"DRAFT_INIT_DRIVER_NAME":     i.driver,
+		"DRAFT_INIT_DRAFT_NAMESPACE": draftNamespace,
+	}
+
+	if i.ingressEnabled {
+		envMap["DRAFT_INIT_INGRESS_ENABLED"] = "1"
+	}
+
+	if i.upgrade {
+		envMap["DRAFT_INIT_UPGRADE"] = "1"
+	}
+
+	for key, val := range envMap {
+		os.Setenv(key, val)
+	}
+}
+
+func (i *initCmd) setupDraftConfig() error {
+	i.draftConfig = &installerConfig.DraftConfig{
+		Image: i.image,
+	}
+	if err := i.tlsOptions(i.draftConfig); err != nil {
+		return err
+	}
+	log.Debugf("raw chart config: %s", i.draftConfig)
+	return nil
+}
+
+func (i *initCmd) prepareDraftChart() error {
+	chartFiles := installer.DefaultChartFiles
+	if tlsEnable {
+		chartFiles = append(chartFiles, &chartutil.BufferedFile{
+			// TODO: Add chartutil.SecretName to k8s.io/helm/pkg/chartutil/create.go
```

Comment:
should we take this TODO out since it's not directly related to draft?

Reply:
:+1: